bot.py

```
@@ -30,41 +82,130 @@ def create_reddit_instance():
+    submission.mod.lock()
+    submission.mod.send_removal_message(type="public_as_subreddit", message=REMOVAL_MESSAGE)
+    submission.mod.send_removal_message(type="private", title=MESSAGE_SUBJECT, message=REMOVAL_MESSAGE)
+
+def process_submissions(subreddit) -> None:
+    """
+    Process all new submissions in the specified subreddit.
+
+    Checks for low-quality images and takes appropriate actions (report or remove).
+    """
+    logging.debug("Start processing submissions...")
+    num_processed = 0
     num_reports = 0
-
-    # Iterate through new submissions
-    for s in subreddit.new(limit=NUM_SUBMISSIONS):
-        timestamp = datetime.fromtimestamp(int(s.created_utc))
-
-        if s.link_flair_text not in ["Question", "Success Story!", "Meta"]:
-            image_width_px = extract_image_width(s.selftext)
-            resolution = convert_px_to_dpi(image_width_px)
-
-            if image_width_px < IMAGE_WIDTH_THRESHOLD_PX:
-                num_reports += 1
-                s.report(f"low quality image: {resolution}DPI")
-                print(f"{timestamp} {resolution}DPI {s.author} id: {s.id}")
-
-    print(f"number of submissions reported: {num_reports}")
+    num_removed = 0
+
+    for submission in subreddit.new(limit=None):
+
+        # only check unmoderated posts
+        if submission.approved:
+            continue
+        
+        num_processed += 1
+        num_comments = submission.num_comments
+        timestamp = datetime.fromtimestamp(int(submission.created_utc))
+
+        if timestamp > CUTOFF_DATE:
+            if submission.link_flair_text not in ["Question", "Success Story!", "Meta"]:
+                image_width_px = extract_image_width(submission.selftext)
+                date = timestamp.date()
+
+                if image_width_px == 0:
+                    logging.error(f"{date} Error: unable to process submission {submission.id} by /u/{submission.author}")
+                    continue
```

since your function is supposed to return the `image_width_px`, i wouldn't the return value to signal that an error occurred. what you can do instead, is have the try/except raise the exception (propagate it up) instead of `return 0`, then you can wrap `image_width_px = extract_image_width(submission.selftext)` in a try/except

---

I'll need to figure out proper commenting here, but I do like that the bot not only handles reasonable but lower than specified image resolution based on letter-sized paper, it also handles resumes on A4 for the international applicants.

---

I agree. Returning 0 here as an error code is not recommended because the function signature and doc string indicate that it's always an int that represents width. I'd raise an exception as well and propagate it, just like @gjbranham mentioned.